plugins/inputs/apcupsd/apcupsd_test.go

```diff
@@ -0,0 +1,129 @@
+package apcupsd
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"net"
+	"sync"
+	"testing"
+
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/testutil"
+)
+
+func TestApcupsdGather(t *testing.T) {
+	input, ok := inputs.Inputs["apcupsd"]
+	if !ok {
+		t.Fatal("Input not defined")
+	}
+	apc := input().(*ApcUpsd)
+
+	apc.Description()
+	apc.SampleConfig()
+
+	one := boolToInt(true)
+	if one != 1 {
+		t.Errorf("boolToInt failed")
+	}
+
+	if testing.Short() {
+		t.Skip("Skipping network dependent tests in short mode")
```

Comment:
It is okay to not skip the network tests so long as they don't require any outside services to be running, and they complete immediately.